### PR TITLE
fix(http) : set response.ok based on given status code

### PR DESCRIPTION
**Pull request by nexus-uw**
- **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  bug fix
- **What is the current behavior?** (You can also link to an open issue here)
  response.ok is not defined
- **What is the new behavior (if this is a feature change)?**
  response.ok is set to true if status is 200-299 (as already documented)

closes #6503

### CODE REVIEW

File: modules/angular2/src/http/static_response.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export class Response {
   constructor(responseOptions: ResponseOptions) {
     this._body = responseOptions.body;
     this.status = responseOptions.status;
+    this.ok = (this.status >= 200 && this.status <= 299);
```
Comment (mhevery) [inline modules/angular2/src/http/static_response.ts]: I think `success` would be a better field name.

**Comment by mhevery** [comment]
Please address the comment. looks good otherwise.

File: modules/angular2/src/http/static_response.ts (hunk 1 above)
Comment (nexus-uw) [inline modules/angular2/src/http/static_response.ts]: @mhevery , I am in agreement with you on this but the spec for fetch's response object has it defined as _ok_ (ref: https://developer.mozilla.org/en-US/docs/Web/API/Response/ok, https://github.com/nexus-uw/angular/blob/http/fix-ok/modules/angular2/src/http/static_response.ts#L21-L25).

Comment (mhevery) [inline modules/angular2/src/http/static_response.ts]: OK, let's follow the spec.

**Pull request closed without merging**